src/gflownet/tasks/seh_frag.py

```diff
@@ -116,83 +118,82 @@ def compute_flat_rewards(self, mols: List[RDMol]) -> Tuple[FlatRewards, Tensor]:
-            "replay_buffer_warmup": 10000,
-            "mp_pickle_messages": False,
-            "algo": "TB",
-        }
+    def set_default_hps(self, cfg: Config):
+        cfg.hostname = socket.gethostname()
+        cfg.pickle_mp_messages = False
+        cfg.num_workers = 8
+        cfg.opt.learning_rate = 1e-4
+        cfg.opt.weight_decay = 1e-8
+        cfg.opt.momentum = 0.9
+        cfg.opt.adam_eps = 1e-8
+        cfg.opt.lr_decay = 20_000
+        cfg.opt.clip_grad_type = "norm"
+        cfg.opt.clip_grad_param = 10
+        cfg.algo.global_batch_size = 64
+        cfg.algo.offline_ratio = 0
+        cfg.model.num_emb = 128
+        cfg.model.num_layers = 4
+
+        cfg.algo.method = "TB"
+        cfg.algo.max_nodes = 9
+        cfg.algo.sampling_tau = 0.9
+        cfg.algo.illegal_action_logreward = -75
+        cfg.algo.train_random_action_prob = 0.0
+        cfg.algo.valid_random_action_prob = 0.0
+        cfg.algo.tb.epsilon = None
+        cfg.algo.tb.bootstrap_own_reward = False
+        cfg.algo.tb.Z_learning_rate = 1e-3
+        cfg.algo.tb.Z_lr_decay = 50_000
+        cfg.algo.tb.do_parameterize_p_b = False
+
+        cfg.replay.use = False
+        cfg.replay.capacity = 10_000
+        cfg.replay.warmup = 1_000
 
     def setup_algo(self):
-        algo = self.hps.get("algo", "TB")
+        algo = self.cfg.algo.method
         if algo == "TB":
             algo = TrajectoryBalance
         elif algo == "FM":
             algo = FlowMatching
+        elif algo == "A2C":
+            algo = A2C
+        elif algo == "SQL":
+            algo = SoftQLearning
         else:
             raise ValueError(algo)
-        self.algo = algo(self.env, self.ctx, self.rng, self.hps, max_nodes=self.hps["max_nodes"])
+        self.algo = algo(self.env, self.ctx, self.rng, self.cfg)
 
     def setup_task(self):
         self.task = SEHTask(
             dataset=self.training_data,
-            temperature_distribution=self.hps["temperature_sample_dist"],
-            temperature_parameters=self.hps["temperature_dist_params"],
+            cfg=self.cfg,
             rng=self.rng,
-            num_thermometer_dim=self.hps["num_thermometer_dim"],
             wrap_model=self._wrap_for_mp,
         )
 
     def setup_model(self):
-        self.model = GraphTransformerGFN(self.ctx, num_emb=self.hps["num_emb"], num_layers=self.hps["num_layers"])
+        model = GraphTransformerGFN(
+            self.ctx,
+            self.cfg,
+            do_bck=self.cfg.algo.tb.do_parameterize_p_b,
+        )
+        self.model = model
 
     def setup_env_context(self):
         self.ctx = FragMolBuildingEnvContext(
-            max_frags=self.hps["max_nodes"], num_cond_dim=self.hps["num_thermometer_dim"]
+            max_frags=self.cfg.algo.max_nodes, num_cond_dim=self.cfg.task.seh.num_thermometer_dim
         )
 
     def setup(self):
-        hps = self.hps
         RDLogger.DisableLog("rdApp.*")
         self.rng = np.random.default_rng(142857)
```

Comment:
I noticed you often pass a rng object. I have the habit of passing random see instead (from which I init an rng object). Any rational for that? Mine is that I always like and prefer to have "serializable" objects in my signatures.

I am just wondering.

Reply:
I'm leaving this for a future PR, but I'd like to improve the way both the rngs and device objects are used and passed around in the code, which right now is messy because I'm dealing with pytorch multiprocessing workers in a hacky way.